### PR TITLE
Add more snapshots to Wallet Mode, fix presentation issue with Apple Pay

### PR DESCRIPTION
**Pull request by davidme-stripe**
## Summary
* Add more snapshot tests
* Fix an issue where the sheet would not show Apple Pay if Apple Pay was the only available payment method.

## Testing
Adds tests for CI

### CODE REVIEW

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/SavedPaymentMethods/SavedPaymentMethodsViewController.swift [hunk 1]
```diff
@@ -120,7 +120,7 @@ class SavedPaymentMethodsViewController: UIViewController {
         self.isApplePayEnabled = isApplePayEnabled
         self.spmsCompletion = spmsCompletion
         self.delegate = delegate
-        if savedPaymentMethods.isEmpty {
+        if savedPaymentMethods.isEmpty && !isApplePayEnabled {
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/SavedPaymentMethods/SavedPaymentMethodsViewController.swift:123]: Great catch, but this means that we need to change line 288:

```
                    return savedPaymentMethods.isEmpty && !isApplePayEnabled ? .close(showAdditionalButton: false) : .back
```

Comment (davidme-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/SavedPaymentMethods/SavedPaymentMethodsViewController.swift]: Done!

**Review by wooj-stripe — APPROVED** [review]
lgtm!

File: StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/SavedPaymentMethods/SavedPaymentMethodsViewController.swift [hunk 2]
```diff
@@ -120,14 +120,14 @@ class SavedPaymentMethodsViewController: UIViewController {
         self.isApplePayEnabled = isApplePayEnabled
         self.spmsCompletion = spmsCompletion
         self.delegate = delegate
-        if savedPaymentMethods.isEmpty {
+        if !savedPaymentMethods.isEmpty || isApplePayEnabled {
```
Comment (wooj-stripe) [inline StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/SavedPaymentMethods/SavedPaymentMethodsViewController.swift:123]: We should probably use `shouldShowPaymentMethodCarousel` here?

Comment (davidme-stripe) [reply StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/SavedPaymentMethods/SavedPaymentMethodsViewController.swift]: Good point, refactored a bit. We can't call it on `self` before `super.init`, but we can at least share the logic.

**Review by wooj-stripe — APPROVED** [review]

**Review by wooj-stripe — DISMISSED** [review]

**Review by wooj-stripe — APPROVED** [review]

**Pull request merged**